docs/managing-dragonfly/operator/prometheus-guide.md

```diff
@@ -0,0 +1,106 @@
+---
+sidebar_position: 4
+---
+
+# Integrate Prometheus with Dragonfly operator
+
+This guide provides a step-by-step instruction to set up Prometheus with Dragonfly
+Operator. In this guide, you're going to use [Promethues Operator](https://github.com/prometheus-operator/prometheus-operator) along with PodMonitor to manage Prometheus
+resources.
+
+## Prerequisites
+
+- A kubernetes cluster with [Dragonfly installed](./installation.md)
+
+## Install Prometheus Operator
+
+First make sure you have all the Prometheus CRDs installed in your cluster. Run the
+below commands to install the Prometheus Operator.
+
+```
+LATEST=$(curl -s https://api.github.com/repos/prometheus-operator/prometheus-operator/releases/latest | jq -cr .tag_name)
+curl -sL https://github.com/prometheus-operator/prometheus-operator/releases/download/${LATEST}/bundle.yaml | kubectl create -f -
+```
+
+## Create the PodMonitor Resource
+
+PodMonitors allow Prometheus to monitor specific pods that has the target labels. It
+is the easiest way to monitor Dragonfly pods. You can either create your own
+PodMonitor config file or use our sample [podMonitor.yaml](https://github.com/dragonflydb/dragonfly-operator/blob/main/monitoring/podMonitor.yaml)
+file to create a PodMonitor object.
+
+```
+kubectl apply -f https://github.com/dragonflydb/dragonfly-operator/blob/main/monitoring/podMonitor.yaml
+```
+
+Note that you must use the `app: <dragonfly-name>` label as a selector label in the PodMonitor in order to target the correct dragonfly instances. The value of `app`
+label is the name of your dragonfly resource name (in this case, `dragonfly-sample`).
+
+Dragonfly pod exposes a port named `admin` and you can use it as the endpoint in PodMonitor.
+
+## Create Promethues Resources
+
+Now that you have installed the operator and pod-monitor, it is time to create
+`prometheus` resources. If you have RBAC enabled, create necessary `serviceaccount`,
+`clusterrole` and `clusterrolebinding` resources first.
+
+```
+$ kubectl apply -f https://github.com/dragonflydb/dragonfly-operator/blob/main/monitoring/promServiceAccount.yaml
+$ kubectl apply -f https://github.com/dragonflydb/dragonfly-operator/blob/main/monitoring/promClusterRole.yaml
+$ kubectl apply -f https://github.com/dragonflydb/dragonfly-operator/blob/main/monitoring/promClusterBinding.yaml
+```
+
+This will allow prometheus to scrape data from dragonfly resources. Run the below
```

Comment:
This will allow Prometheus to scrape data from Dragonfly resources. Run the below command to create the Prometheus object. It will create a pod named
`prometheus-prometheus-0`.